java-to-zod-gradle-plugin/src/main/java/sh/ivan/zod/GenerateZodSchemas.java

```diff
@@ -0,0 +1,321 @@
+    private List<String> excludePropertyAnnotations = Collections.emptyList();
+
+    // Library used in JSON classes (e.g., Jackson2, Gson)
+    private JsonLibrary jsonLibrary;
+
+    // Jackson 2 global configuration
+    private Jackson2Configuration jackson2Configuration;
+
+    // Gson global configuration
+    private GsonConfiguration gsonConfiguration;
+
+    // JSON-B global configuration
+    private JsonbConfiguration jsonbConfiguration;
+
+    // Scan Spring REST application for classes to process
+    private boolean scanSpringApplication = false;
+
+    // Specifies the level of logging output (Debug, Verbose, etc.)
+    private Logger.Level loggingLevel = Logger.Level.Verbose;
+
+    // Skip processing
+    private boolean skip = false;
+    private List<String> optionalAnnotations = Collections.emptyList();
+
+    public GenerateZodSchemas() {}
+
+    @TaskAction
+    public void runTask() {
+        try {
+            init();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public void init() throws IOException {
+        TypeScriptGenerator.setLogger(new Logger(loggingLevel));
+        TypeScriptGenerator.printVersion();
+        if (skip) {
+            TypeScriptGenerator.getLogger().info("Skipping plugin execution");
+            return;
+        }
+
+        // class loader
+        List<URL> urls = new ArrayList<>();
+        // Add the output directory for your compiled classes (this excludes dependencies)
+        File classesDir = new File(getBuildDirectoryAsFile(), "classes/java/main");
+
+
+        // Check if the directory exists before adding it
+        if (classesDir.exists()) {
+            urls.add(classesDir.toURI().toURL());
+        }
+
+        // Create a class loader using URLs from the classpath
+        try (URLClassLoader classLoader = new URLClassLoader(
+                urls.toArray(new URL[0]), Thread.currentThread().getContextClassLoader())) {
+
+            Settings settings = createSettings(classLoader);
+
+            cz.habarta.typescript.generator.Input.Parameters parameters =
+                    new cz.habarta.typescript.generator.Input.Parameters();
+            parameters.classNames = classes;
+            parameters.classNamePatterns = classPatterns;
+            parameters.classesWithAnnotations = classesWithAnnotations;
+            parameters.classesImplementingInterfaces = classesImplementingInterfaces;
+            parameters.classesExtendingClasses = classesExtendingClasses;
+            parameters.jaxrsApplicationClassName = classesFromJaxrsApplication;
+            parameters.automaticJaxrsApplication = classesFromAutomaticJaxrsApplication;
+            parameters.isClassNameExcluded = settings.getExcludeFilter();
+            parameters.classLoader = classLoader;
+            parameters.scanningAcceptedPackages = scanningAcceptedPackages;
+            parameters.debug = loggingLevel == Logger.Level.Debug;
```

Comment:
And we can probably share all of this config/setup once we have an abstract class for all the params